### ISSUE TITLE
(2006, "MySQL server has gone away (error(32, 'Broken pipe'))")

### ISSUE THREAD

**Issue by emamirazavi** [coleifer/peewee#538]
When i want to save my model, my connection broke! and raises an exception with title of this issue, but mysql logfile reads:
INSERT INTO `inboundmsg` (`tojid`, `fromjid`, `type`, `datetime`, `content`, `url`, `size`, `local`, `status`) VALUES ('989393310102', '989102260264', 'image', '2015-02-26 19:43:11.789153', '/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCABKAGQDASIAAhEBAxEB/8QAHwAAAQUBAQEBAQEAAAAAAAAAAAECAwQFBgcICQoL/8QAtRAAAgEDAwIEAwUFBAQAAAF9AQIDAAQRBRIhMUEGE1FhByJxFDKBkaEII0KxwRVS0fAkM2JyggkKFhcYGRolJicoKSo0NTY3ODk6Q0RFRkdISUpTVFVWV1hZWmNkZWZnaGlqc3R1dnd4eXqDhIWGh4iJipKTlJWWl5iZmqKjpKWmp6ipqrKztLW2t7i5usLDxMXGx8jJytLT1NXW19jZ2uHi4+Tl5ufo6erx8vP09fb3+Pn6/8QAHwEAAwEBAQEBAQEBAQAAAAAAAAECAwQFBgcICQoL/8QAtREAAgECBAQDBAcFBAQAAQJ3AAECAxEEBSExBhJBUQdhcRMiMoEIFEKRobHBCSMzUvAVYnLRChYkNOEl8RcYGRomJygpKjU2Nzg5OkNERUZHSElKU1RVVldYWVpjZGVmZ2hpanN0dXZ3eHl6goOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4uPk5ebn6Onq8vP09fb3+Pn6/9oADAMBAAIRAxEAPwDz22+K15Y2i21jahURQqtJMztgflUA+JN9ePGl7a2kgjyUDKcjPvnNb1p4Y0e78T6bpFnbIryNh2xn8zWr49+Hmk22u2wtYwgwATK2xGXnOeOuRU+2jJuI50fZw55Ir6B4xfV5FtY7Rllx2YspA9M8gj8RXpelagEtJC0rqVA4KkfnXD6NpdtbacYNOkihTISSVFAAPUDd3OAeldZopSFyMmUFArIzevTB6YIrlm02c103oXIWstVkdhBALtRsJTkHI9ccGs+x8MvpsVxOkELu2XELg7uv8/aqw8R6X4cMttBuiJZmJxk/UH25/SqEHjGS+fy4fMmVGV87sEDPJ9/Wtoo1VJW1FntYdZ0+VUkjtrsSbWSYYPX0xmsq8QCyhF/cs0yv8209D05P4ZxzXYahbR67ZrJax7NUjbcSp2vL7ZHf3rmx4Y1a+QzXNvcvbwSFGM4Clm5zjoTjHNVFW0MZwaMmQpuYeZHORwMnkA+39Ks6dLbRh4XJbIO87tuPpUdv4ZvILGXXN0YtLfDGOY43ewzycZql9pjZpg5U7v8AVbX4UkngnrxSlEzSOqtIrGfDi5AgUryAMjHb3611Vpdw2jvDAxZV4JVRwPrXl1pZmC/vI7a4W5g2I25flJJXkD2Bzg10lpdQQssNtHcbwdrt5mS5B4O3t+dYyitildHqcMU13BHNCdyso/i6H0NFddoUCtpVu0oHmMoLfL3orVQVjoUD578NWkJ8dxywptNvcKikHqOhruPjBY6XP5NrcPFFeSwN5bv1xnkCuI03Tru08c2slv5rIZ8yRqM9+tbnx30nXda1TTItAVpQlq8kwB+b7w2gDvn5vyrhwjcrns5rBKEbHll5LY6fbR2M3nQxct58nKyem3b0OK1f7T/srRZGt55X8xUMZGQWXB5+b6Yz71BoVv48+02+njTZJEVsGK4t1TA6H5mAwORzWrN4R13x1o19drcpbR6ZI8OwoyhnGAyDPXp1/Cutqz948WlBuSseb3niea4hfeY2nZsfOchR24rPh1K7jmUiR5COiodoPvUOt+FvEMetNC+l3Ad2yhVCVx9elbdx4fl8PeFnup7iOTUbicRxRlc4CjLADuSSBntj3rR1qcbJPc640Kkru2iN7TfHJto1KuVlTb5mDkYHHTrXr8dhqGraXb3SXbpFJGZUAIUsGHXkjg9enNeX+Evh/wCRp9vqGrwpcX98qi2t3GyGFs/dbn5m9q6zWdVbw5NaLqkFvLM8jRrHb/MRj7x+gwOPcVN1U+Exq03HRnR3d9p2k+HJtBvZ+ZUO8RMJGAbucZrjLb4bagJfN00q8DxrIDMfJOD04J4rofDtzY2t7Lq0N5eWf2yIQSzeRuEJDZHXjkHGfUD1rsPttnql28kBunlS3d5Gl2+cqFsBUYcLggnBz1oU7S5GhPDp0+dM4vRPAl4l1LHfS6bFHwdqXQMgOO49MV6L4Q0rTvDDNIzQSNPwdqEkAfd5P/Av0rn4/h617cyXb38tz5jnY8kOGjUDgM2Rzz9OK6aDw7pmnXiX2ppbPIjAW0e0M3HTn06VScY+8QqMr2O4+0rgbeBjpRXEX/iLFy3O3POM0VHtDsVIbp9vBBq6TOgBOQxrF+I3h/UNbmjuNCcW2prCEiumfYoXnK7vetjw039pGMZ3uvfv0rb1Oyt45IJL9naOElhbglgzDkMBn9Kyp0pUbqXc1rVo1bNPocj4Y0/UdE0mU6/LBJrM2FWRXz5iIBt9OeefXg1Ym8Rmws2N1ExhJLNGFHJ74I710J0aDxPbSyXdpqNjJuzG0xVT9QAens1ZnjfQdTi8Lxz2cEeoahax7JolOwToBjdzxnAyR35qZqbvJbGcOVaPc8Q+LXxCitrc/wBmoMyDahx0968ZstTu9c8Q2Kxxy3N20ipFGMkkk9ABXrNlOmqeK4bfUNIa03Dy5baaDzFlQnBPI+VgMcj0rr9L8F+A/Dc8XiuK+Nl5ayQfZZJNjGdcggA9CfyGQaKEYRTSV2zWtUnK13ZI77wXpMFzo6W99L5k0ADAFExC/Q4J6/U+nFc54o8GWGj6dq+r2trcT6iJxma8bKqpPJiA4we561J4c8eeHo7d/Kg86GYBZHyzbSclhlvvDJ5x0GKzfjD4502LwHc2Gl3Ucl1KFEdvE+9gNwP4cetVGXJNQREo+0g5s5jT/GqWd7a210mJJ28tAoGAff2rv9N8d2NnFKTp8QkIDOVUAMemTgdeK+TbbU57jUFe5uBAwPDOQpX8+/vXrXwn1aKLxjZt9qF5IWCvgFo0HuenHXNejJcyPMS5Xoez2mu+Idet5DYWUtvbSDIuGQrGoPcH+I/SszVF1HSI7o6jc+fOjA7l4X5uQEPfgHP/ANatHxl4i1jRNF1aO9sfK0U2ZitZdOcOfNYkYGBkDByPpWbANSvdEtrDUAgtTaJJbeYfMlaQgklyRxj09/auCpqtWehShyvQ5a41tppmcuefeiuXZjG7ozAFWIPPvRWB06HdeH9T1DSbmO4sNSLbf4J0EikenY/rXb2fiuS9uxNerHCSAGMBIzj68gfTn3ryO3usDritG1vyMYevfq0Y1FaR4lOrKDuj1+PxjdrMUtNHkmgU/wCs89AW+i5zWtpvjPTLy4FrMXtLpuBDcLsZvoD1/CvIrTVzH8rE9OtaaamkuwsFcodylhnB9RXE8HJfBI6ViYv4onXeLfB8RuF1rQ43EkakzW0GP3ox1UHjPsMZ7Ed/OPD/AIXtL/wnr8vjCzvGiYEQeapjZHUttaInqSNuc/3ee9d5pni+4h2q0isPRhWf4s1P7fp8yXSJJYzqYp1BIkUN/ErA9QcEemKhYSSdyvrEWj5g8Y3E9vqdlBayvEi2/lrBnAVSOfTqQc/SrfhbRmv8Ehrg93DEIPxHB/AVsfEXwVeWraWRvvUnbyLa4VCfOO75VfHRueR717x4M0jR4PhvFp17JEJIEctKkTKI3AxkEgFjkf5B53lJQMYwclY8G1/wpctCJBZadMiKQGklZHHtmsb4caXe614gisl1V9MtVmRXiSRgzqWwVU+o6fjXoHiR47qzb7Ncb4mfZ5qnKlkbt+I/WuW0fXjp9/Iq6QJDbqXS5UbQHQsAGPTkAe9VLbQzje9me06zHeqbiSO7jvdEt5xDZwtHlwyqf4upxjgnmsp7e6uPCsQ0+48u+Zmmk3MQQhBO3/PvWb4H8d6vqsMlv4fhs4rmICaWEqDnnGFJGAfQ96tajpfjEeINP1L7FdT6bPua5ZiiMit0+XIIxz2rhcbS0O5Sbi7s8gvL6eO5dd3eipfGFnLpuv3Vo67GiO3HXPvnvmikoo1uzpzMAOtC3QUfe+nNUiflqB+n417s1qeSzdh1L5gsjY9DV631QRzBS3B6HNcbcE7hyauEnbEc84qBI7eW/wAJuQkkc1K2pG7sJofMIZ0IFc/ETheT0qO3JAbBI+Y0Ad58NNYa+07ZcyJLLa5dYpVGN6nk/XaGwfevR7y6+1eHo5dMQ3OlXStlVdU8oDoV4wy5BGDk9OK8N+FZJ8QopJKtdMCOxBJyK9S0N3t/EGtWtuzRWsdzHshQ7UXLc4UcDNcGIjaR24d3ieKeOJktfE2r6NpsAncvFLBDbIWbc6HcoQcg5AyO2a3PDng68062ubDxEhtH1OMSBd4dolOQdw6Bgece1c54cd2/aG1CQsxkF1L8xPP3fWvSvFBLeONHjYlo206Msp5BJZskislUekPIdSklF1PM8dg0PU/Cni2eKwjn1bTXVhJNZQl1kjboCQDtNfUPhay/tTwzbxGW4t7t7dXeKVd8kQ7DcBkH2JzW34LtoIIJI4IYo41AwqIAB+AqeaWRby5CyOAJIAAGPGSc0laWo/I861H4LWuqXH2xrqRmkUEmaPa34gjIor0++J87r2/xoqfYRL9vI//Z', 'https://----/d/xOX7wcwGDoX13Mmk-7xHZlTvRhcABRAAB0IQjQ/An-vJIOekxRv77ykjdGy6LOTtwk_j-3GcAq1VBeQ-pNt.jpg', 57180, '5dce934c-bdd2-11e4-b641-6c8814f85df8.jpg', '3')

and has not any error!!!
any idea about peewee? peewee is being boring for me...

I use playhouse to make pool and this is my pool instance:
my_db = PooledMySQLDatabase(dbconfig['database'], host=dbconfig['host'], 
                            port=dbconfig['port'], user=dbconfig['user'], 
                            passwd=dbconfig['password'], threadlocals=True, max_connections=None,
                            charset='utf8mb4')
also i use flask for web interface

**Comment by coleifer** [comment]
http://docs.peewee-orm.com/en/latest/peewee/database.html#error-2006-mysql-server-has-gone-away

**Comment by coleifer** [comment]
For an alternate fix, see also 017e4e4952f0b429dd21f0e90465a0f644cf6015

**Comment by GreatBahram** [comment]
For a new alternate fix:
```python
from playhouse.shortcuts import RetryOperationalError


class MyRetryDB(RetryOperationalError, MySQLDatabase):
    pass
```
